core/controllers/admin.py

```diff
@@ -865,22 +992,29 @@ class AdminRoleHandler(base.BaseHandler):
     }
 
     @acl_decorators.can_access_admin_page
-    def get(self):
-        filter_criterion = self.normalized_request.get(
-            'filter_criterion')
+    def get(self) -> None:
+        assert self.user_id is not None
+        request_data = cast(
+            AdminRoleHandlerNormalizedGetRequestDict,
+            self.normalized_request
+        )
+        filter_criterion = request_data['filter_criterion']
         if filter_criterion == feconf.USER_FILTER_CRITERION_ROLE:
-            role = self.normalized_request.get(
-                feconf.USER_FILTER_CRITERION_ROLE)
+            role = request_data[feconf.USER_FILTER_CRITERION_ROLE]
             role_services.log_role_query(
                 self.user_id, feconf.ROLE_ACTION_VIEW_BY_ROLE,
                 role=role)
             self.render_json({
-                'usernames': user_services.get_usernames_by_role(role)
+                'usernames': (
+                    user_services.get_usernames_by_role(role) if role else []
+                )
             })
         elif filter_criterion == feconf.USER_FILTER_CRITERION_USERNAME:
-            username = self.normalized_request.get(
-                feconf.USER_FILTER_CRITERION_USERNAME)
-            user_id = user_services.get_user_id_from_username(username)
+            username = request_data[feconf.USER_FILTER_CRITERION_USERNAME]
+            user_id = (
+                user_services.get_user_id_from_username(username)
+                if username else None
```

Comment:
None `username` is ok here because in the next line we are passing the username to `role_services.log_role_query` method https://github.com/oppia/oppia/blob/900a25a62a37057c4779247cf9c9c0f14b66e788/core/controllers/admin.py#L884 

and in `log_role_query` method we are passing Optional[str] `username` directly to `RoleQueryAuditModel` https://github.com/oppia/oppia/blob/900a25a62a37057c4779247cf9c9c0f14b66e788/core/domain/role_services.py#L309

and in RoleQueryAuditModel, `username` can be None https://github.com/oppia/oppia/blob/900a25a62a37057c4779247cf9c9c0f14b66e788/core/storage/audit/gae_models.py#L56

But None `user_id` is not acceptable, and for this we already have a check https://github.com/oppia/oppia/blob/900a25a62a37057c4779247cf9c9c0f14b66e788/core/controllers/admin.py#L887

Reply:
oh sorry, I misinterpreted the code. Now, we are throwing an error for the None username.